Add OnlyExplicitMappedMembers implementation

Kataane opened the pull request:
# Add OnlyExplicitMappedMembers Configuration Option

## Description

First, I want to express my gratitude for the beautiful library and the excellent work that has been done in Mapperly. This library has been incredibly useful!

This PR introduces a new `OnlyExplicitMappedMembers` configuration option to the `MapperAttribute` that allows mapping only properties with explicit configurations (via attributes like `MapProperty`). All other properties will be ignored by default when this option is enabled.

This feature is particularly useful when working with classes that have many properties but you only need to map a few specific ones, providing a more declarative and maintainable approach compared to explicitly ignoring every unwanted property.

__Changes made:__

- Added `OnlyExplicitMappedMembers` property to `MapperAttribute`
- Added corresponding configuration to `MapperConfiguration` and merger logic
- Implemented the ignore logic in `IgnoredMembersBuilder` to automatically ignore non-matching members when this option is enabled
- Added comprehensive unit tests covering different scenarios (extra source members, extra target members, and extra members in both)
- Updated test infrastructure to support the new configuration option

Fixes #1981

## Checklist

- [x] The existing code style is followed
- [x] The commit message follows our guidelines
- [x] Performed a self-review of my code
- [x] Hard-to-understand areas of my code are commented
- [ ] The documentation is updated (as applicable)
- [x] Unit tests are added/updated
- [ ] Integration tests are added/updated (as applicable, especially if feature/bug depends on roslyn or framework version in use)

Kataane committed:
Cleanup and improve OnlyExplicitMappedMembers implementation

latonz commented on test/Riok.Mapperly.Tests/Mapping/ObjectPropertyIgnoreTest.cs:
```diff
@@ -168,4 +168,73 @@ public void WithNestedIgnoredSourceAndTargetPropertyShouldDiagnostic()
     }
+
+    [Fact]
+    public void OnlyExplicitMappedMembersWithExtraSourceMembers()
```
Also add tests with explicit mappings (MapProperty, MapPropertyFromSource, ...)

latonz reviewed (changes requested):
Thank you for your contribution, I added my feedback.

latonz commented:
Closing for now as no update was received for several weeks... Feel free to reopen with updates.

Kataane commented:
Addressed all review feedback: fixed the ignore logic to only map members with explicit configurations (MapProperty/MapPropertyFromSource), updated tests to cover explicit mapping scenarios, and fixed the XML doc comment to use a see cref tag.

Kataane commented:
Addressed all review feedback: fixed the ignore logic to only map members with explicit configurations (MapProperty/MapPropertyFromSource), added tests covering explicit mapping scenarios, fixed the XML doc to use see cref, and rebased onto current main.
